game-core/src/main/java/games/strategy/triplea/delegate/power/calculator/MainDefenseCombatValue.java

```diff
@@ -0,0 +1,138 @@
+package games.strategy.triplea.delegate.power.calculator;
+
+import games.strategy.engine.data.GameData;
+import games.strategy.engine.data.GamePlayer;
+import games.strategy.engine.data.TerritoryEffect;
+import games.strategy.engine.data.Unit;
+import games.strategy.triplea.Constants;
+import games.strategy.triplea.attachments.RulesAttachment;
+import games.strategy.triplea.attachments.UnitSupportAttachment;
+import games.strategy.triplea.delegate.TerritoryEffectHelper;
+import java.util.Collection;
+import java.util.function.Predicate;
+import lombok.AccessLevel;
+import lombok.Builder;
+import lombok.Getter;
+import lombok.NonNull;
+import lombok.Value;
+
+/**
+ * Calculates defense strength and roll for normal dice
```

Comment:
Is there a low luck variant of this class? "normal dice" makes me think there is a second defense combat value used for LL.

Reply:
roll and strength are the same for LL and non-LL.  So, no there isn't a low luck variant.  This is for "non-AA" dice but I didn't want to say that.  Any ideas on a better name?

Reply:
Perhaps: `Calculates defense strength and roll bonuses`

Reply:
Only problem I have with that is that these classes don't just calculate the bonus.  They calculate the final strength and roll value.  That includes adding the bonuses as well as limiting the value within the dice sides.  I've changed the text to say `Calculates defense strength and roll for non-AA dice` for now.